### PR TITLE
Identify JSX Spread Attributes as Expression parents

### LINKED ISSUE #4688: Variable references not detected when used in a spread (in a JSX element)
The underlying problem I'm experiencing is described [in this comment](https://github.com/palantir/tslint/issues/570#issuecomment-138332733)

And I think the following test, if my attempts to use the other `tsxRename*.ts` files as a template has succeeded and my understanding of what `tslint` is doing with `getDocumentHighlights` is correct, also highlights the problem:

```typescript
/// <reference path='fourslash.ts' />

//@Filename: file.tsx
//// declare module JSX {
////     interface Element { }
////     interface IntrinsicElements {
////     }
////     interface ElementAttributesProperty { props }
//// }
//// class MyClass {
////   props: {
////     name?: string;
////     size?: number;
////   }
//// }
////
//// var [|nn|]: {name?: string; size?: number};
//// var x = <MyClass {...[|n/**/n|]}></MyClass>;

goTo.marker();
verify.renameLocations(false, false);
```

Does that make sense?

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #4688

### CODE REVIEW

File: tests/cases/fourslash/jsxSpreadReference.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+////     name?: string;
+////     size?: number;
+////   }
+//// }
+////
+//// var [|/*dst*/nn|]: {name?: string; size?: number};
+//// var x = <MyClass {...[|n/*src*/n|]}></MyClass>;
+
+goTo.marker('src');
+debugger;
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/jsxSpreadReference.ts]: # No

File: tests/cases/fourslash/jsxSpreadReference.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+goTo.definition();
+verify.caretAtMarker('dst');
+
+goTo.marker('src');
+verify.renameLocations(false, false);
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/jsxSpreadReference.ts]: Can you add the param comments for `false` and `false`?

File: tests/cases/fourslash/jsxSpreadReference.ts (hunk 1 above)
Comment (danquirk) [inline tests/cases/fourslash/jsxSpreadReference.ts]: :ticket:

Comment (RyanCavanaugh) [inline tests/cases/fourslash/jsxSpreadReference.ts]: :disappointed:

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**